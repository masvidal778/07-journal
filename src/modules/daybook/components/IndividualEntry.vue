<script>
import {defineComponent} from 'vue'

export default defineComponent({
  name: "IndividualEntry",
  props: {
    entry: {
      type: Object,
      required: true
    }
  },
  computed: {
    shortText() {
      return( this.entry.text.length  > 130)
          ? this.entry.text.substring(0, 130) + '...'
          : this.entry.text
    }
  }
})
</script>

<template>

  <div
      class="entry-container mb-3 pointer p-2"
      @click="$router.push( {name: 'entry', params: { id: 10 }} )"
  >
    <!-- titol -->
    <div class="entry-title d-flex">
      <span class="text-success fs-5 fw-bold">15</span>
      <span class="mx-1 fs-5">Setembre</span>
      <span class="mx-2 fw-light">2023, divendres</span>
    </div>

    <div class="entry-description">
      {{ shortText }}
    </div>
  </div>

</template>

<style lang="scss" scoped>
.entry-container{
  border-bottom: 1px solid #2c3e50;
  transition: 0.2s all ease-in;

  &:hover{
    background-color: lighten($color: grey, $amount: 45);
    transition: 0.2s all ease-in;
  }

  .entry-description{
    font-size: 12px;
  }

}




</style>