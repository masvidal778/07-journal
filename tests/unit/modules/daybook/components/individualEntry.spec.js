import { shallowMount } from "@vue/test-utils";
import IndividualEntry from "@/modules/daybook/components/IndividualEntry.vue";
import { journalState } from "../../../mock-data/test-journal-state";

describe('Testing individualEntry Component', () => {

    //mockRouter
    const mockRouter = {
        push: jest.fn()
    }

    //const wrapper = shallowMount ( entry, { props... global... } )
    const wrapper = shallowMount( IndividualEntry, {
        props: {
            entry: journalState.entries[0]
        },
        global: {
            mocks: {
                $router: mockRouter
            }
        }
    })

    test('should match snapshot', () => {

        expect( wrapper.html() ).toMatchSnapshot()
    })

    test('if click to entry-container should redirect to', () => {

        const entryContainer = wrapper.find('.entry-container')
        entryContainer.trigger('click')

        expect(mockRouter.push).toHaveBeenCalledWith(
            {
                name: 'entry',
                params: {
                    id: journalState.entries[0].id
                }
            }
        )

    })

    test('testing computed properties', () => {

        // wrapper.vm <------ computed properties
        // day: 18
        // month: Juliol
        // yearDay: '2021, Diumenge'
        expect(wrapper.vm.day).toBe(18)
        expect(wrapper.vm.month).toBe('Juliol')
        expect(wrapper.vm.yearDay).toBe('2021, Diumenge')

    })

})