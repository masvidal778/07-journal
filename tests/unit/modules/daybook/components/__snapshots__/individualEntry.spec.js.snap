// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing individualEntry Component should match snapshot 1`] = `
<div class="entry-container mb-3 pointer p-2">
  <!-- titol -->
  <div class="entry-title d-flex"><span class="text-success fs-5 fw-bold">18</span><span class="mx-1 fs-5">Juliol</span><span class="mx-2 fw-light">2021, Diumenge</span></div>
  <div class="entry-description">Aliquam tristique erat massa, at commodo mauris vulputate vel. Aenean ut tincidunt magna, quis bibendum risus. Etiam dignissim, ip...</div>
</div>
`;
